runtime/vm/JFRConstantPoolTypes.hpp

```diff
@@ -1246,6 +1274,63 @@ class VM_JFRConstantPoolTypes {
 		gcConfiguration->heapAddressBits = J9JAVAVM_REFERENCE_SIZE(vm) * 8;
 	}
 
+	static uintptr_t processNativeLibrariesCallback(const char *libraryName, void *lowAddress, void *highAddress, void *userData)
+	{
+		VM_JFRConstantPoolTypes *constantPoolTypes = reinterpret_cast<VM_JFRConstantPoolTypes *>(userData);
+		J9Pool *nativeLibrariesTable = constantPoolTypes->_nativeLibrariesTable;
+		NativeLibraryEntry *firstNativeLibraryEntry = constantPoolTypes->_firstNativeLibraryEntry;
+		NativeLibraryEntry *previousNativeLibraryEntry = constantPoolTypes->_previousNativeLibraryEntry;
+		NativeLibraryEntry *entry = firstNativeLibraryEntry;
+		PORT_ACCESS_FROM_JAVAVM(constantPoolTypes->_vm);
+		char *libraryNameCopy = (char *)j9mem_allocate_memory(strlen(libraryName) + 1, OMRMEM_CATEGORY_VM);
+		if (libraryNameCopy == NULL) {
+			/* Allocation for library name failed. */
+			return 1;
+		}
```

Comment:
This is too early; if an existing entry is found, the allocated memory is leaked. This should go just before line 1302.

Please capture the name length in a local rather than call `strlen()` a second time on line 1313, and use the length in a call to `memcpy()` rather than use `strcpy()`.